### PR TITLE
Move ex_004_nonnegative.jl to the new framework

### LINKED ISSUE #491: DG tests that need to be moved to the new framework
I discussed with @lcw and @jkozdon which DG methods tests/examples need to be moved to the new framework so that we can finally kill the old one. I marked those that have been moved but not yet merged. 

- [ ] test/DGmethods_old/conservation/sphere.jl
- [x] test/LinearSolvers/poisson.jl
- [x] test/DGmethods_old/util/filter_test.jl
- [x] test/DGmethods_old/util/nonnegative_test.jl
- [x] test/DGmethods_old/sphere/advection_sphere_*
- [ ] examples/DGmethods_old/ex_004_nonnegative.jl

The last two can be combined into one test and I will try to do that this week.

Additionally, the following could be moved but there is no functionality to take local gradient of auxiliary state in the new framework.
- ~~test/DGmethods_old/util/grad_test.jl~~ (not grad aux in new framework)
- ~~test/DGmethods_old/util/grad_test_sphere.jl~~  (not grad aux in new framework)

@simonbyrne

### PR DESCRIPTION
**Pull request by mwarusz**
- Adds upwind numerical flux to the advection-diffusion model
- Moves ex_004_nonnegative.jl to the new framework

Together with #985 will close #491

# For review by CLIMA developers

- [ ] There are no open pull requests for this already
- [ ] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://climate-machine.github.io/CLIMA/latest/CodingConventions.html) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Comment by mwarusz** [comment]
bors try

**Comment by jkozdon** [comment]
@mwarusz is this ready to go? Do you want me to fix the merge conflicts?

File: examples/Numerics/DGmethods/nonnegative.jl [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    @__DIR__,
+    "..",
+    "..",
+    "..",
+    "test",
+    "Numerics",
+    "DGmethods",
+    "advection_diffusion",
+    "advection_diffusion_model.jl",
+))
```
Comment (simonbyrne) [inline examples/Numerics/DGmethods/nonnegative.jl:47]: Unfortunately this broke the docs, as the relative path is no longer correct for the notebook generated by Literate.jl.

Comment (charleskawczynski) [reply examples/Numerics/DGmethods/nonnegative.jl:47]: Hm. Also, does it seem awkward for a tutorial to use code in `test/`?

Comment (simonbyrne) [reply examples/Numerics/DGmethods/nonnegative.jl:47]: Yes, we should probably have a better way of doing that.

Comment (jkozdon) [reply examples/Numerics/DGmethods/nonnegative.jl:47]: We could make the advection diffusion a first class physics model(?)

Comment (jkozdon) [reply examples/Numerics/DGmethods/nonnegative.jl:47]: Is there a way in Julia to get base path of the project / git repo?

Comment (charleskawczynski) [reply examples/Numerics/DGmethods/nonnegative.jl:47]: Following `using CLIMA`: `const clima_dir = dirname(dirname(pathof(CLIMA)))`

Comment (charleskawczynski) [reply examples/Numerics/DGmethods/nonnegative.jl:47]: I really like the idea of making advection-diffusion a first class physics model, though. I think abstracting some of the models will help make them more easily extendable.

Comment (jkozdon) [reply examples/Numerics/DGmethods/nonnegative.jl:47]: > We could make the advection diffusion a first class physics model(?)

Alternatively, advection-diffusion could be turned in a tutorial on how to create a physics model.

**Pull request merged**